src/Concerns/ManagesCustomer.php

```diff
@@ -261,47 +263,161 @@ public function syncStripeCustomerDetails()
     }
 
     /**
-     * The discount that applies to the customer, if applicable.
+     * The discount that applies to the customer's primary subscription, if applicable.
      *
      * @return \Laravel\Cashier\Discount|null
      */
     public function discount()
     {
-        $customer = $this->asStripeCustomer(['discount.promotion_code']);
+        // Since customer-level discounts are no longer supported, check any active subscription
+        // Try default subscription first, then any active subscription
+        $subscription = $this->subscription() ?: $this->subscriptions->where('stripe_status', 'active')->first();
 
-        return $customer->discount
-            ? new Discount($customer->discount)
-            : null;
+        if (! $subscription) {
+            return null;
+        }
+
+        // Use the same expansion logic as the subscription's discount method
+        $stripeSubscription = $subscription->asStripeSubscription(['discounts.promotion_code']);
+
+        if (isset($stripeSubscription->discounts) && ! empty($stripeSubscription->discounts)) {
+            return new Discount($stripeSubscription->discounts[0]);
+        }
+
+        return null;
     }
 
     /**
-     * Apply a coupon to the customer.
+     * Get all discounts that apply to the customer's subscriptions.
+     *
+     * @return \Illuminate\Support\Collection|\Laravel\Cashier\Discount[]
+     */
+    public function discounts()
+    {
+        // Since customer-level discounts are no longer supported, collect discounts from all subscriptions
+        return $this->subscriptions()->map(function ($subscription) {
+            return $subscription->discounts();
+        })->flatten();
+    }
+
+    /**
+     * Apply a coupon to the customer's subscriptions.
+     * By default, applies to the primary subscription only for safety.
      *
      * @param  string  $coupon
+     * @param  string|array|null  $subscriptionTypes  Specific subscription types to target, or null for primary only
      * @return void
+     *
+     * @throws \Laravel\Cashier\Exceptions\InvalidCoupon
      */
-    public function applyCoupon($coupon)
+    public function applyCoupon($coupon, $subscriptionTypes = null)
```

Comment:
some method use `$coupon` while other use `$couponId`, might be best to standardise it.